sdk/storage/Azure.Storage.Files.Shares/tests/ServiceClientTests.cs

```diff
@@ -145,6 +145,36 @@ await foreach (Page<ShareItem> page in service.GetSharesAsync().AsPages())
             Assert.IsTrue(shares.All(c => c.Properties.Metadata == null));
         }
 
+        [Test]
+        public async Task ListSharesSegmentAsync_Premium()
+        {
+            // Arrange
+            ShareServiceClient service = GetServiceClient_Premium();
+            string shareName = GetNewShareName();
+
+            // Ensure at least one premium share
+            await using DisposingShare test = await GetTestShareAsync(
+                service: service,
+                shareName: shareName);
+            ShareClient share = test.Share;
+
+            var shares = new List<ShareItem>();
+            await foreach (Page<ShareItem> page in service.GetSharesAsync().AsPages())
+            {
+                shares.AddRange(page.Values);
+            }
+
+            // Assert
+           ShareItem premiumShareItem = shares.Where(r => r.Name == shareName).First();
+            Assert.IsNotNull(premiumShareItem.Properties.ETag);
```

Comment:
Rather than just operating on the first ShareItem, should we use shares.All in the assertions?

Reply:
There's no reason these properties would be returned for 1 premium file share but not the other premium file shares.  The feature either works or it doesn't.

Reply:
What if there was a deserialization bug that caused only the first shareItem to be populated?

Reply:
I'm filtering by name, which is a unique GUID.  There should be only one ShareItem.